astropy/cosmology/flrw.py

```diff
@@ -141,120 +141,85 @@ def __init__(self, H0, Om0, Ode0, Tcmb0=0.0*u.K, Neff=3.04, m_nu=0.0*u.eV,
         self.m_nu = m_nu  # (reset later, this is just for unit validation)
         self.Ob0 = Ob0  # (must be after Om0)
 
-        # Derived quantities
+        # Derived quantities:
+        # Dark matter density; matter - baryons, if latter is not None.
         self._Odm0 = None if Ob0 is None else (self._Om0 - self._Ob0)
 
         # 100 km/s/Mpc * h = H0 (so h is dimensionless)
-        self._h = self._H0.value / 100.
+        self._h = self._H0.value / 100.0
         # Hubble distance
         self._hubble_distance = (const.c / self._H0).to(u.Mpc)
-        # H0 in s^-1; don't use units for speed
+        # H0 in s^-1
         H0_s = self._H0.value * H0units_to_invs
-        # Hubble time; again, avoiding units package for speed
-        self._hubble_time = u.Quantity(sec_to_Gyr / H0_s, unit=u.Gyr)
+        # Hubble time
+        self._hubble_time = sec_to_Gyr / H0_s << u.Gyr
 
-        # critical density at z=0 (grams per cubic cm)
+        # Critical density at z=0 (grams per cubic cm)
         cd0value = critdens_const * H0_s ** 2
-        self._critical_density0 = u.Quantity(cd0value, unit=u.g / u.cm ** 3)
+        self._critical_density0 = cd0value << u.g / u.cm ** 3
 
-        # -------------------
-        # neutrinos
+        # Compute photon density from Tcmb
+        self._Ogamma0 = a_B_c2 * self._Tcmb0.value ** 4 / self._critical_density0.value
 
-        # Load up neutrino masses.
-        self._nneutrinos = floor(self._Neff)
-        self._neff_per_nu = np.nan  # assuming no neutrinos
-
-        # Start by assuming massless neutrinos, then correct below.
-        self._massivenu = False
-        self._massivenu_mass = None
-        self._nmassivenu = 0
-        self._nmasslessnu = self._nneutrinos
-
-        # We are going to share Neff between the neutrinos equally.
-        # In detail this is not correct, but it is a standard assumption
-        # because properly calculating it is a) complicated b) depends
-        # on the details of the massive neutrinos (e.g., their weak
-        # interactions, which could be unusual if one is considering sterile
-        # neutrinos)
-        if self._nneutrinos > 0 and self._Tcmb0.value > 0:
-            self._neff_per_nu = self._Neff / self._nneutrinos
+        # Compute Neutrino temperature:
+        # The constant in front is (4/11)^1/3 -- see any cosmology book for an
+        # explanation -- for example, Weinberg 'Cosmology' p 154 eq (3.1.21).
+        self._Tnu0 = 0.7137658555036082 * self._Tcmb0
```

Comment:
I'd tend to just put `(4/11)**(1/3)` here - I don't understand trying to optimize for speed here (but I realize this is just being refactored, so fine not to do anything like this in this PR!)